src/components/modals/TeachSessionModal.tsx

```diff
@@ -673,6 +679,7 @@ class TeachModal extends React.Component<Props, ComponentState> {
         let saveDisable = !this.state.hasTerminalAction 
                         || this.props.teachSession.dialogMode === CLM.DialogMode.Extractor
                         || this.props.teachSession.botAPIError !== null
+        this.state.promptOnAbandon = !saveDisable
```

Comment:
This doesn't seem to be fixing what the bug is - namely don't ask abandon if there is no data yet.  This seems to remove the prompt for other unrelated reasons (i.e. in extractor mode) which doesn't make sense.  Let's talk